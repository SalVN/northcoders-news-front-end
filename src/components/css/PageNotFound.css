.page-not-found-card {
    margin: 20px;
    text-align: center;
}

.page-not-found-title {
    color: #c1002e;
    font-weight: 700;
    font-size: 20px;
}

.page-not-found-button {
    background-color: #c1002e;
    color: white;
    border-radius: 20px;
    border: 0px;
    margin-top: 20px;
}