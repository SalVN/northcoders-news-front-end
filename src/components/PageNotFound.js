import React from 'react';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Link } from 'react-router-dom';

import './css/PageNotFound.css';

const PageNotFound = function (props) {
    return (
        <div className='card page-not-found-card'>
            <div className='card-image'>
                <img src='http://res.cloudinary.com/dl37xtqhv/image/upload/v1505237157/error-2129569_1920_wflvai.jpg'/>
            </div>
            <div className='card-content'>
                <div className='page-not-found-title'>Page Not Found</div>
                <div>That page does not exist</div>
                <Link to='/'><button className='button page-not-found-button'>Return to home page</button></Link>
            </div>
        </div>
    );
};

PageNotFound.propTypes = {

};

export default PageNotFound;