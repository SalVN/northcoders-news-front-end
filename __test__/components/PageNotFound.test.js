import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';


import PageNotFound from '../../src/components/PageNotFound';

describe('PageNotFound', () => {
    it('is a function', () => {
        expect(typeof PageNotFound).toEqual('function');
    });

    it('renders', () => {
        const enzymeWrapper = shallow(<PageNotFound />);
        expect(enzymeWrapper.children().length).toEqual(2);
    });

    it('renders correctly', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <PageNotFound />
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains one link', () => {
        const enzymeWrapper = shallow(<PageNotFound />);
        expect(enzymeWrapper.find('Link').length).toBe(1);
    });
});