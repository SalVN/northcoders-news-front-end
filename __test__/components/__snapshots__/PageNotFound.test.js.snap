// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound renders correctly 1`] = `
<div
  className="card page-not-found-card"
>
  <div
    className="card-image"
  >
    <img
      src="http://res.cloudinary.com/dl37xtqhv/image/upload/v1505237157/error-2129569_1920_wflvai.jpg"
    />
  </div>
  <div
    className="card-content"
  >
    <div
      className="page-not-found-title"
    >
      Page Not Found
    </div>
    <div>
      That page does not exist
    </div>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="button page-not-found-button"
      >
        Return to home page
      </button>
    </a>
  </div>
</div>
`;
